includes/admin/class-wp-job-manager-admin-notices.php

```diff
@@ -161,27 +168,48 @@ public static function init_core_notices() {
 	}
 
 	/**
-	 * Dismiss notices as requested by user. Inspired by WooCommerce's approach.
+	 * Dismiss a notice.
 	 */
 	public static function dismiss_notices() {
 		if ( isset( $_GET['wpjm_hide_notice'] ) && isset( $_GET['_wpjm_notice_nonce'] ) ) {
-			if ( ! wp_verify_nonce( wp_unslash( $_GET['_wpjm_notice_nonce'] ), 'job_manager_hide_notices_nonce' ) ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Nonce should not be modified.
+			if ( ! wp_verify_nonce( wp_unslash( $_GET['_wpjm_notice_nonce'] ), 'job_manager_notices_nonce' ) ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Nonce should not be modified.
 				wp_die( esc_html__( 'Action failed. Please refresh the page and retry.', 'wp-job-manager' ) );
 			}
 
 			if ( ! current_user_can( 'manage_options' ) ) {
 				wp_die( esc_html__( 'You don&#8217;t have permission to do this.', 'wp-job-manager' ) );
 			}
 
-			$hide_notice = sanitize_key( wp_unslash( $_GET['wpjm_hide_notice'] ) );
+			$notice_id = sanitize_key( wp_unslash( $_GET['wpjm_hide_notice'] ) );
 
-			self::remove_notice( $hide_notice );
+			self::dismiss_notice( $notice_id );
 
-			wp_safe_redirect( remove_query_arg( [ 'wpjm_hide_notice', '_wpjm_notice_nonce' ] ) );
+			wp_safe_redirect( remove_query_arg( [ 'wpjm_hide_notice', '_wpjm_notice_nonce', 'wpjm_notice_action' ] ) );
 			exit;
 		}
 	}
 
+	/**
+	 * Call an action when a notice button is clicked.
+	 */
+	public static function handle_notice_action() {
+
+		if ( isset( $_GET['wpjm_notice_action'] ) && isset( $_GET['_wpjm_notice_nonce'] ) ) {
+
+			$action = sanitize_key( wp_unslash( $_GET['wpjm_notice_action'] ) );
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Nonce should not be modified.
+			if ( ! wp_verify_nonce( wp_unslash( $_GET['_wpjm_notice_nonce'] ), 'job_manager_notices_nonce' ) ) {
+				wp_die( esc_html__( 'Action failed. Please refresh the page and retry.', 'wp-job-manager' ) );
+			}
+
+			if ( ! current_user_can( 'manage_options' ) ) {
+				wp_die( esc_html__( 'You don&#8217;t have permission to do this.', 'wp-job-manager' ) );
+			}
+
+			do_action( 'job_manager_action_' . $action );
```

Comment:
This is a pretty common pattern in WordPress and WPJM, like `wp_ajax_*`, `admin_action_*`. A bit harder to search but cleaner and faster to use.

Reply:
Yeah whenever I encountered it in the existing cases it always needed extra time to figure out. It usually goes like this:
- You use the hotkey to go to the hook definition. This doesn't work because the IDE can't figure out dynamic hooks.
- Then I search for the whole name
- Then I search for parts of the hook. It is also very common during this step for the results to be too many as the parts of the hook name are very commonly used for other things like variables or function names.

I don't see how dynamic hooks are cleaner as instead of doing concatenation we will pass an extra argument. The only downside is an extra check on the called method which I don't think is important enough.